x-pack/plugins/ml/public/application/services/usage_collection.test.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { UsageCollectionSetup } from 'src/plugins/usage_collection/public';
+
+import { mlUsageCollectionProvider } from './usage_collection';
+
+describe('usage_collection', () => {
+  let usageCollection: UsageCollectionSetup | null = null;
+
+  beforeEach(() => {
+    usageCollection = ({
+      reportUiCounter: jest.fn(),
+    } as unknown) as UsageCollectionSetup;
+  });
+
+  test('mlUsageCollection', () => {
+    const mlUsageCollection = mlUsageCollectionProvider(
+      (usageCollection as unknown) as UsageCollectionSetup
+    );
+
+    mlUsageCollection.click('test');
+    mlUsageCollection.count('test');
+    // @ts-expect-error reportUiCounter is a mock function
+    expect(usageCollection.reportUiCounter.mock.calls).toHaveLength(2);
```

Comment:
```suggestion
    expect(usageCollection.reportUiCounter).toHaveBeenCalledTimes(2);
```
but you should also assert arguments, so please consider `toHaveBeenCalledWith`

Reply:
Changed in https://github.com/elastic/kibana/pull/107772/commits/b622cfd2a80be91a587e2db557bdc983c6f934a6

Reply:
Great! have you tried `toHaveBeenCalledWith` to assert the actual calls?

Reply:
Added extra tests in https://github.com/elastic/kibana/pull/107772/commits/45640bc3d29fd6c8d3afdf81943d88f6248929d4